source/ddbus/thin.d

```diff
@@ -175,6 +175,19 @@ InterfaceName interfaceName(string path) pure @nogc nothrow @safe {
   return cast(InterfaceName) path;
 }
 
+/// Serving as a typesafe alias for a FileDescriptor.
+enum FileDescriptor : int {
```

Comment:
FileDescriptors are _unsigned_ 32 bit numbers according to spec

Change to uint and check if this breaks anything (for example with none, -1 being equal to uint.max then)

Reply:
I've modified the FileDescriptor to be an uint, but I'm not sure if I should keep the `FileDescriptor.none`. 

I believe I'd put it in, since in C a file descriptor is usually represented by a signed integer, with -1 being returned by `open` and such in case of errors. The standard hard and soft limits of file descriptors per process on Linux are not high enough to come even close to `uint.max`, so the special value would not be a practical problem.

My use case for it was if a misbehaving (C) application were to send an invalid file descriptor over dbus, the user could do a simple check with `if (fd != FileDescriptor.none)` to check if the file descriptor is valid. Although it could very well be that dbus itself or anything else is already enforcing this.

Anyway, it seems I'm trying to find an excuse for why I wrote this code a month ago :smile: , right now I think I should remove it due to it very likely having no use at all.